azblob/url_container.go

```diff
@@ -89,7 +86,7 @@ func (c ContainerURL) Create(ctx context.Context, metadata Metadata, publicAcces
 // For more information, see https://docs.microsoft.com/rest/api/storageservices/delete-container.
 func (c ContainerURL) Delete(ctx context.Context, ac ContainerAccessConditions) (*ContainerDeleteResponse, error) {
 	if ac.IfMatch != ETagNone || ac.IfNoneMatch != ETagNone {
-		panic("the IfMatch and IfNoneMatch access conditions must have their default values because they are ignored by the service")
+		sanityCheckFailed("the IfMatch and IfNoneMatch access conditions must have their default values because they are ignored by the service")
```

Comment:
This function already has an error on the return signature. Why not return an error here rather than a panic?

Reply:
The discussion will continue [here](https://github.com/Azure/azure-storage-blob-go/pull/73#discussion_r222069003).